<?xml version="1.0"?>
<launch>
    <!-- config file path  -->
    <arg name="rviz_settings" default="$(find amr_system)/config/rviz/local_map_creator.rviz"/>
    <arg name="obstacle_detector_settings" default="$(find amr_system)/config/param/obstacle_detector.yaml"/>
    <arg name="local_map_creator_settings" default="$(find amr_system)/config/param/local_map_creator.yaml"/>

    <!-- node launch -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_settings)"/>
    <node pkg="amr_system" type="obstacle_detector_node" name="obstacle_detector" output="screen">
        <rosparam command="load" file="$(arg obstacle_detector_settings)"/>
    </node>
    <node pkg="amr_system" type="local_map_creator_node" name="local_map_creator" output="screen">
        <rosparam command="load" file="$(arg local_map_creator_settings)"/>
    </node>
    <node pkg="rosbag" type="play" name="play" args="$(env HOME)/bagfiles/obstacle_detector.bag"/>
</launch>
